control-plane/api-gateway/binding/validation.go

```diff
@@ -204,11 +208,59 @@ func validateTLS(gateway gwv1beta1.Gateway, tls *gwv1beta1.GatewayTLSConfig, res
 	return nil, err
 }
 
+// Envoy will silently reject any keys that are less than 2048 bytes long
+// https://github.com/envoyproxy/envoy/blob/main/source/extensions/transport_sockets/tls/context_impl.cc#L238
+const MinKeyLength = 2048
+
 func validateCertificateData(secret corev1.Secret) error {
-	_, _, err := common.ParseCertificateData(secret)
+	_, privateKey, err := common.ParseCertificateData(secret)
 	if err != nil {
 		return errListenerInvalidCertificateRef_InvalidData
 	}
+
+	err = validateKeyLength(privateKey)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func validateKeyLength(privateKey string) error {
+	// we can assume this is non-nil as it would've caused an error from common.ParseCertificate if it was
+	privateKeyBlock, _ := pem.Decode([]byte(privateKey))
+
+	if privateKeyBlock.Type != "RSA PRIVATE KEY" {
```

Comment:
Is there a constant for this in the std library?

Reply:
not that I've found 
https://pkg.go.dev/crypto/rsa#pkg-constants
https://pkg.go.dev/crypto/x509#pkg-constants